lib/core/base/check.js

```diff
@@ -180,21 +180,37 @@ Check.prototype.runSync = function(node, options, context) {
  */
 
 Check.prototype.configure = function(spec) {
+	// allow test specs (without evaluate functions) to work as
+	// internal checks
+	if (!spec.evaluate || metadataFunctionMap[spec.evaluate]) {
+		this._internalCheck = true;
+	}
+
 	if (spec.hasOwnProperty('enabled')) {
 		this.enabled = spec.enabled;
 	}
 
 	if (spec.hasOwnProperty('options')) {
-		this.options = normalizeOptions(spec.options);
+		// only normalize options for internal checks
+		if (this._internalCheck) {
+			this.options = normalizeOptions(spec.options);
+		} else {
+			this.options = spec.options;
+		}
 	}
 
 	['evaluate', 'after']
 		.filter(prop => spec.hasOwnProperty(prop))
 		.forEach(prop => (this[prop] = createExecutionContext(spec[prop])));
 };
 
-Check.prototype.getOptions = function getOptions(options = {}) {
-	return deepMerge(this.options, normalizeOptions(options));
+Check.prototype.getOptions = function getOptions(options) {
+	// only merge and normalize options for internal checks
+	if (this._internalCheck) {
```

Comment:
I think we allow deep merge for any non-array object

```suggestion
	// only merge and normalize options for internal checks
	if (typeof this.options === 'object' && Array.isArray(this.options)) {
```